src/mapgen/mg_biome.cpp

```diff
@@ -192,7 +192,14 @@ BiomeGenOriginal::BiomeGenOriginal(BiomeManager *biomemgr,
 
 	heatmap  = noise_heat->result;
 	humidmap = noise_humidity->result;
+
 	biomemap = new biome_t[m_csize.X * m_csize.Z];
+	// Initialise with the ID of the default biome, so that cavegen and
+	// dungeongen can get biomes when biome generation (which calculates the
+	// biomemap IDs) is disabled.
+	for (s32 i = 0; i < m_csize.X * m_csize.Z; i++) {
+		biomemap[i] = 0;
```

Comment:
the 0 is a non sense here, it should be nullptr instead as it's a pointer, we are not in C93 :D

Reply:
in fact you can do a better thing here and more performant:

memset(biomemap, 0, sizeof(biome_t) * m_csize.X * m_csize.Z);

Reply:
Ok.

Reply:
Actually 0 is intentional, see the comment above, it is the biome ID of the default biome, nullptr would cause a crash.
Will use the suggested code.

Reply:
Actually, i prefer to keep it more readable, it runs on startup only so isn't performance critical. Thanks for the suggestion though.

Reply:
@nerzhul Taking the current code for your suggestion: `biomemap` is a pointer but gets dereferenced by indexing it, so setting the value to `nullptr` is nonsense.
However, I prefer to use `memset` here since no further processing (which requires a loop) is needed here.
```C++
biomemap = new biome_t[m_csize.X * m_csize.Z];
// Initialize with the default biome ID. Used after processing in cavegen and dungeongen
memset(biomemap, 0, sizeof(biome_t) * m_csize.X * m_csize.Z);
```

Reply:
Ok will do.

Reply:
Done.